Documentation Updates `[AARD-2027]` (#1285)
Autodesk/synthesis: fission/README.md

# Fission

> Fission is Synthesis' web-based robotics simulator. This app is hosted [on our website](https://synthesis.github.com/fission/), in addition to a closed [beta version](https://synthesis.autodesk.com/beta/).

Reviewer: Is this something that users can request? Or just exclusive invite. If it's the former than it would be good to add some instructions (perhaps link to a google form if there is one)

Author: We should probably have an avenue for people to request the closed beta, or just make it open.

## Setup & Building

### Requirements

1. Bun (v1.2.0 recommended)

>    - Yarn, Bun, or any other package managers work just as well.

Reviewer 2: ```suggestion
   - NPM, Yarn, or any other package managers work just as well.
```

> You can either run the `init` command or run the following commands details in "Specific Steps":

Reviewer 3: ```suggestion
You can either run the `init` command or run the following commands detailed in "Specific Steps":
```

To install all dependencies:

```bash
bun i
```

> For the asset pack that will be available in production, download the asset pack [here](https://synthesis.autodesk.com/Downloadables/assetpack.zip) and unzip it.

Reviewer 2: The assetpack script no longer uses the production assetpack url.

We have two packaging commands: one for compiling dev for attachment to the in-development endpoint, and another for the release endpoint.

Release:

```bash
bun run build:prod
```

In-development:

```bash
bun run build:dev
```

You can alternatively run the default build command for your own hosting:

```bash
bun run build
```

### Electron Packaging

We also give you the option to package Synthesis with electron. This will not give a performance boost, but it will allow Synthesis to work offline (make sure to also launch the app and download all the robot/field files you want to use).

To package the app run:

```bash
bun run electron:publish
```

The packaged app will be located in the `/fission/out` directory.

## Core Systems

> These core systems make up the lion's share of the fission source code. Each systems manages a different aspect of the simulated world

Reviewer 3: ```suggestion
These core systems make up the lion's share of the fission source code. Each system manages a different aspect of the simulated world
```

| `host`               | Starts the development server used for testing and exposes it to the network. Supports hot-reloading (though finicky with WASM module loading). |
| `dev`                | Starts the development server used for testing. Supports hot-reloading (though finicky with WASM module loading).                               |
| `test`               | Runs the unit tests via Vitest.                                                                                                                 |
| `build`              | Builds the project into its packaged form. Uses the root base path.                                                                             |
| `build:prod`         | Builds the project into its packaged form. Uses the `/fission/` base path.                                                                      |
| `build:dev`          | Builds the project into its packaged form. Uses the `/fission-closed/` base path.                                                               |
| `preview`            | Runs the built project for preview locally before deploying.                                                                                    |

> | `lint`               | Runs lints the projects with Biome.                                                                                                             |

Reviewer 4: ```suggestion
| `lint`               | Runs lint on the project with Biome.                                                                                                             |
```

> | `lint:fix`           | Attempts to fix issues found with Biome lintign.                                                                                                |

Reviewer 3: ```suggestion
| `lint:fix`           | Attempts to fix issues found with Biome linting.                                                                                                |
```

We also give you the option to package Synthesis with electron. This will not give a performance boost, but it will allow Synthesis to work offline (make sure to also launch the app and download all the robot/field files you want to use).

To package the app run:

```bash
npm run electron:publish
```

The packaged app will be located in the `/fission/out` directory.

## Core Systems

> These core systems make up the lionshare of the fission source code. Each systems manages a different aspect of the simulated world

Reviewer 3: ```suggestion
These core systems make up the lion's share of the fission source code. Each system manages a different aspect of the simulated world
```

Reviewer 2: ```suggestion
These core systems make up most of the fission source code. Each system manages a different aspect of the simulated world
```

Is it even true that these are >50% of it?

- [World](/fission/src/systems/World.md)
- [Scene Renderer](/fission/src/systems/scene/SceneRenderer.md)
- [Physics System](/fission/src/systems/physics/PhysicsSystem.md)
- [Input System](/fission/src/systems/input/InputSystem.md)
- UI System

## Additional Systems

These systems will extend off of the core systems to build out features in Synthesis.

> - [Simulation System](fission/src/systems/simulation/SimulationSystem.md)

Reviewer 3: ```suggestion
- [Simulation System](/fission/src/systems/simulation/SimulationSystem.md)
```

| `build:prod`         | Builds the project into its packaged form. Uses the `/fission/` base path.                                        |
| `preview`            | Runs the built project for preview locally before deploying.                                                      |
| `lint`               | Runs ESLint on the project.                                                                                       |
| `lint:fix`           | Attempts to fix issues found with ESLint.                                                                         |
| `prettier`           | Runs Prettier on the project as a check.                                                                          |
| `prettier:fix`       | Runs Prettier on the project to fix any issues with formatting.                                                   |
| `format`             | Runs `prettier:fix` and `lint:fix`.                                                                               |
| `assetpack`          | Downloads the assetpack and unzips/installs it in the correct location.                                           |

> | `playwright:install` | Downloads the Playwright browsers.                                                                                |

Reviewer 3: Should Electron script be added here? Ex `electron:make`

We also give you the option to package Synthesis with electron. This will not give a performance boost, but it will allow Synthesis to work offline (make sure to also launch the app and download all the robot/field files you want to use).

To package the app run:

```bash
npm run electron:publish
```

The packaged app will be located in the `/fission/out` directory.

## Core Systems

These core systems make up the lion's share of the fission source code. Each systems manages a different aspect of the simulated world

- [World](/fission/src/systems/World.md)
- [Scene Renderer](/fission/src/systems/scene/SceneRenderer.md)
- [Physics System](/fission/src/systems/physics/PhysicsSystem.md)
- [Input System](/fission/src/systems/input/InputSystem.md)
- UI System

> ## Additional Systems

Reviewer 2: does this distinction still make sense

| `test`               | Runs the unit tests via Vitest.                                                                                   |
| `build`              | Builds the project into its packaged form. Uses the root base path.                                               |
| `build:prod`         | Builds the project into its packaged form. Uses the `/fission/` base path.                                        |
| `preview`            | Runs the built project for preview locally before deploying.                                                      |
| `lint`               | Runs ESLint on the project.                                                                                       |
| `lint:fix`           | Attempts to fix issues found with ESLint.                                                                         |
| `prettier`           | Runs Prettier on the project as a check.                                                                          |
| `prettier:fix`       | Runs Prettier on the project to fix any issues with formatting.                                                   |

> | `format`             | Runs `prettier:fix` and `lint:fix`.                                                                               |

Reviewer 2: These aren't accurate any more and this list isn't comprehensive
